src/abc_atlas_access/abc_atlas_cache/cloud_cache.py

```diff
@@ -432,6 +430,98 @@ def __init__(self,
         # last loaded from this cache dir (if applicable)
         self._manifest_last_used = c_path / '_manifest_last_used.txt'
 
+        # self._downloaded_data_path is where we will keep a JSONized
+        # dict mapping paths to downloaded files to their file_hashes;
+        # this will be used when determining if a downloaded file
+        # can instead be a symlink
+        self._downloaded_data_path = c_path / '_downloaded_data.json'
+
+        # if the local manifest is missing but there are
+        # data files in cache_dir, emit a warning
+        # suggesting that the user run
+        # self.construct_local_manifest
+        if not self._downloaded_data_path.exists():
+            file_list = c_path.glob('**/*')
+            has_files = False
+            for fname in file_list:
+                # If the file exists ignore it if it is a json file
+                # the last used manifest file, or a .DS_Store file for Mac
+                # compatibility.
+                if fname.is_file() \
+                        and 'json' not in fname.name \
+                        and '_manifest_last_used' not in fname.name \
+                        and 'DS_Store' not in fname.name:
+                    has_files = True
+                    break
+            if has_files:
+                msg = 'This cache directory appears to '
+                msg += 'contain data files, but it has no '
+                msg += 'record of what those files are. '
+                msg += 'You should consider running\n\n'
+                msg += f'{self.ui}.construct_local_manifest()\n\n'
+                msg += 'to avoid needlessly downloading duplicates '
+                msg += 'of data files. Running this method will assume that '
+                msg += 'all files currently in the cahce were properly '
+                msg += 'downloaded. To avoid this warning in the future, make '
+                msg += 'sure that\n\n'
+                msg += f'{str(self._downloaded_data_path.resolve())}\n\n'
+                msg += 'is not deleted between instantiations of this '
+                msg += 'cache.'
+                warnings.warn(msg, MissingLocalManifestWarning)
+
+    def construct_local_manifest(self) -> None:
```

Comment:
What do you think about removing the construct but warning users that the missing file will mean that files are potentially going to be re-downloaded? If we remove this, there is a snag in the logic that needs to get moved around. The `_check_successful_download` method is part of `_file_exists` so if local cache for what ever reason lost it's downloaded list it would not return paths to the user. Let me know if this changes your thought process above, if not I can move around the logic a bit to make it safer.

Reply:
The warning you propose is a good idea.

What you say is a good point, but it doesn't change my thinking. Can the classes that inherit from `BasicCloudCache` (or `CloudCacheBase`) carry a boolean indicating if they are local or not.

If local and `_downloaded_data_path` doesn't exist, skip the successful download check.

Most of the data files we are dealing with will be corrupted such that they cannot be opened if they were partially downloaded (HDF5 does a check at the top to make sure the on-disk size matches a file size recorded somehwere in its metadata), so I don't think users will get surprised by loading and analyzing corrupted data.

Reply:
Done this and even found an edge case that was previously not accounted for where a user could use "force_download", have the download fail, and the system would still think the file was downloaded successfully.